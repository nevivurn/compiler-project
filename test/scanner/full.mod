//
// keywords
//
module
procedure
function
extern
var
const
longint
integer
boolean
char
begin
end
if
then
else
while
do
return
true
false


//
// operators
//
:=
+
-
*
/
||
&&
!


//
// relops
//
=
#
<
<=
>=
>


//
// syntax elements
//
,
;
:
.
(
)
[
]


//
// identifiers
//
variable
_temp1


//
// numbers
//
0
1
7375
22575


//
// boolean
//
true
false


//
// characters
//
'a'
'\n'
'\t'


//
// strings
//
"hello, world!"
"string\twith\\escape\ncharacters"
