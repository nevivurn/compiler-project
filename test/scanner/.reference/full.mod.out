scanning 'full.mod'...
  4:1: tModule
  5:1: tProcedure
  6:1: tFunction
  7:1: tExtern
  8:1: tVarDecl
  9:1: tConstDecl
  10:1: tLongint
  11:1: tInteger
  12:1: tBoolean
  13:1: tChar
  14:1: tBegin
  15:1: tEnd
  16:1: tIf
  17:1: tThen
  18:1: tElse
  19:1: tWhile
  20:1: tDo
  21:1: tReturn
  22:1: tBoolConst (true)
  23:1: tBoolConst (false)
  29:1: tAssign
  30:1: tPlusMinus (+)
  31:1: tPlusMinus (-)
  32:1: tMulDiv (*)
  33:1: tMulDiv (/)
  34:1: tOr
  35:1: tAnd
  36:1: tNot
  42:1: tRelOp (=)
  43:1: tRelOp (#)
  44:1: tRelOp (<)
  45:1: tRelOp (<=)
  46:1: tRelOp (>=)
  47:1: tRelOp (>)
  53:1: tComma
  54:1: tSemicolon
  55:1: tColon
  56:1: tDot
  57:1: tLParens
  58:1: tRParens
  59:1: tLBrak
  60:1: tRBrak
  66:1: tIdent (variable)
  67:1: tIdent (_temp1)
  73:1: tNumber (0)
  74:1: tNumber (1)
  75:1: tNumber (7375)
  76:1: tNumber (22575)
  82:1: tBoolConst (true)
  83:1: tBoolConst (false)
  89:1: tCharConst (a)
  90:1: tCharConst (\n)
  91:1: tCharConst (\t)
  97:1: tStringConst ("hello, world!")
  98:1: tStringConst ("string\twith\\escape\ncharacters")
  99:1: tEOF


Done.
